### PR TITLE
(NOBIDS) Improve error handling

### PR DESCRIPTION
**Pull request by recy21**
### <samp>🤖 Generated by Copilot at 582c1e7</samp>

Refactored the error logging in various handler functions to use the `utils.LogError` function for consistency and simplicity. This function wraps the `logger.WithError` function and adds some additional information to the log message. Also removed some unused code and improved the logging fields in `handlers/user.go`.

### CODE REVIEW

File: handlers/pricing.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func MobilePricing(w http.ResponseWriter, r *http.Request) {
 func PricingPost(w http.ResponseWriter, r *http.Request) {
 	err := r.ParseForm()
 	if err != nil {
-		logger.Errorf("error parsing form: %v", err)
+		utils.LogError(err, "error parsing form", 0)
 		utils.SetFlash(w, r, "pricing_flash", "Error: invalid form submitted")
 		logger.Errorf("error parsing pricing request form for %v route: %v", r.URL.String(), err)
```
Comment (D13ce) [inline handlers/pricing.go:118]: What about this line? Why are there two error logs anyways?

File: handlers/user.go [hunk 2]
```diff
@@ -191,9 +192,14 @@ func UserAuthorizeConfirm(w http.ResponseWriter, r *http.Request) {
...
-
 	if err != nil {
-		logger.Errorf("error app not found: %v: %v: %v", user.UserID, appData, err)
+		logger.WithFields(
+			logrus.Fields{
+				"user.UserID": user.UserID,
+				"appData":     appData,
+				"redirectURI": redirectURI,
+			},
+		).WithError(err).Errorf("error app not found")
```
Comment (D13ce) [inline handlers/user.go:202]: Why not use `utils.LogError` here too? See the `additionalInfos` parameter.

**Review by D13ce — COMMENTED** [review]
Left two comments, please address.

File: handlers/pricing.go (hunk 1 above)
Comment (recy21) [reply handlers/pricing.go:118]: I don't know, just changed to utils for better location of the error.
Already asked in Discord to remove the "error parsing form" completely, as I see no point in the info.

File: handlers/user.go (hunk 2 above)
Comment (recy21) [reply handlers/user.go:202]: Currently utils.LogError handles the Fields different, we have to merge the other commit first, before using it for good.

**Review by D13ce — APPROVED** [review]
lgtm

**Pull request merged**